CRM/Contribute/Form/Contribution.php

```diff
@@ -651,92 +651,32 @@ public function buildQuickForm() {
 
     $this->add('select', 'from_email_address', ts('Receipt From'), $this->_fromEmails);
 
-    $status = CRM_Contribute_PseudoConstant::contributionStatus();
-
-    // suppressing contribution statuses that are NOT relevant to pledges (CRM-5169)
-    $statusName = CRM_Contribute_PseudoConstant::contributionStatus(NULL, 'name');
-    if ($this->_ppID) {
-      foreach (array(
-                 'Cancelled',
-                 'Failed',
-                 'In Progress',
-               ) as $suppress) {
-        unset($status[CRM_Utils_Array::key($suppress, $statusName)]);
-      }
-    }
-    elseif ((!$this->_ppID && $this->_id) || !$this->_id) {
-      $suppressFlag = FALSE;
-      if ($this->_id) {
-        $componentDetails = CRM_Contribute_BAO_Contribution::getComponentDetails($this->_id);
-        if (CRM_Utils_Array::value('membership', $componentDetails) || CRM_Utils_Array::value('participant', $componentDetails)) {
-          $suppressFlag = TRUE;
-        }
+    $isUpdate = FALSE;
+    $component = 'contribution';
+    if ($this->_id) {
+      $isUpdate = TRUE;
+      $componentDetails = CRM_Contribute_BAO_Contribution::getComponentDetails($this->_id);
+      if (CRM_Utils_Array::value('membership', $componentDetails)) {
+        $component = 'membership';
       }
-      if (!$suppressFlag) {
-        foreach (array(
-                   'Overdue',
-                   'In Progress',
-                 ) as $suppress) {
-          unset($status[CRM_Utils_Array::key($suppress, $statusName)]);
-        }
-      }
-      else {
-        unset($status[CRM_Utils_Array::key('Overdue', $statusName)]);
+      elseif (CRM_Utils_Array::value('participant', $componentDetails)) {
+        $component = 'participant';
       }
     }
+    elseif ($this->_ppID) {
+      $component = 'pledge';
+    }
```

Comment:
seems to be a problem here where we set `$component` as `pledge` only when the first if block is not `TRUE` which checks whether we are in update mode - `$this->_id`. So, I think `$component` will not be `pledge` even if we are updating any pledge payment? For eg: 

1. Record a payment for pledge with `pending` status.
2. Click `Record Payment` again for the above pending payment. Shouldn't `$component` be equal to `pledge` here?